### DISCUSSION TITLE
When to save settings?

### DISCUSSION THREAD

**Discussion by squivix** [Benji377/OpenMusic#97] [Q&A] [answered]
I was wondering about where we should save the settings. I can think of 2 approaches:

1- A save button in the settings. If the user presses it, all settings are saved. If they leave without pressing it they get a pop-up asking them if they want to save or revert. If we do go with this, should it be in the settings activity? What about the library paths setting that launches a new activity? Should that have its own save button or should it wait for settings activity to save? 

2- App saves settings automatically every time changes are made. Some user prefer this as it usually means less buttons to press.

**Comment by Benji377** [comment, marked as answer]
Personally, I would prefer the second option. Not only are there fewer things to worry about for a user, but we also don't need to worry about where to place the button. The design is also cleaner.

(Sorry I just can't get this design thing out of my head hahah)
